### DISCUSSION TITLE
allow images to be crawled with robots.txt

### DISCUSSION THREAD

**Discussion by kamil-malinski** [directus/directus#15687] [Archive]
Hi!

My frontend serves images like this:

_https://my-directus-domain.com/assets/some-asset-id?key=content-image-850_

But since the robots.txt at https://my-directus-domain.com/robots.txt disallows crawling of the whole domain the google-crawler can not crawl the images either.

Is there a way to change the robots.txt? Creating a robots.txt in /app/dist/ doesn't do anything.

Cheers

Kamil

**Comment by br41nslug** [comment]
The `robots.txt` is static and not extensible currently im afraid.
https://github.com/directus/directus/blob/main/api/src/constants.ts#L63

**Comment by azrikahar** [comment]
We are currently tracking the customization of the `robots.txt` content via env var in this issue: #15421 👍

**Comment by alexreid** [comment]
I commented on issue https://github.com/directus/directus/issues/15421 there, but just in case it's the wrong place, this is a major pain point for our use of Directus Cloud – it affects really common things like setting social metadata images via the CMS, and also causes a lot of errors to be reported in the Google Search Console for pages where we have images being served from Directus.

**Comment by rapliandras** [comment]
I would not go down the rabbit hole of having to reconfigure docker or rebuild stuff inside of it just because of this.
So the simplest solution in my opinion is to solve it on nginx level.

```
location = /robots.txt {
    add_header Content-Type text/plain;
    return 200 "User-agent: *\nAllow: /assets/\nDisallow: /\n";
}
```

**Reply by rijkvanzanten** [reply]
Or you can just set the `ROBOTS_TXT` environment variable 👍🏻  https://directus.io/docs/configuration/general
